hal/interfaces/FlashHomogeneous.cpp

```diff
@@ -17,18 +18,22 @@ namespace hal
     template<class T>
     uint32_t FlashHomogeneousBase<T>::SizeOfSector(T sectorIndex) const
     {
+        assert(sectorIndex < numberOfSectors);
         return sizeOfEachSector;
     }
 
     template<class T>
     T FlashHomogeneousBase<T>::SectorOfAddress(T address) const
     {
-        return address / sizeOfEachSector;
+        auto sectorIndex = address / sizeOfEachSector;
+        assert(sectorIndex <= numberOfSectors);
+        return std::clamp<T>(sectorIndex, 0, numberOfSectors);
```

Comment:
Why use std::clamp here? For debug, there's already an assertion. For release, if someone puts in a wrong address there's no need to make thing look a little less dirty